Library/Homebrew/formula.rb

```diff
@@ -353,18 +353,21 @@ def full_installed_specified_name
 
   # Is the currently active {SoftwareSpec} a {#stable} build?
   # @private
+  sig { returns(T::Boolean) }
   def stable?
     active_spec == stable
   end
 
   # Is the currently active {SoftwareSpec} a {#head} build?
   # @private
+  sig { returns(T::Boolean) }
   def head?
     active_spec == head
   end
 
   # Is this formula HEAD-only?
   # @private
+  sig { returns(T.nilable(T::Boolean)) }
```

Comment:
IMO this should be fixed to not be nilable. Any ? method that's a nilable bool is likely a mistake.